### DISCUSSION TITLE
Cannot edit commit message once files are uploaded

### DISCUSSION THREAD

**Discussion by rushabhT3** [community/community#55123] [Programming Help] [answered]
Shouldn't this be allowed?

**Comment by queenofcorgis** [comment]
Thanks for posting in the GitHub Community, @rushabhT3 ! 

We’ve moved your post to our new Programming Help 🧑‍💻 category, which is more appropriate for this type of discussion.

Please review our [guidelines](https://github.com/orgs/community/discussions/51384) about the Programming Help category for more information.

**Comment by kevinmatthes** [comment, marked as answer]
@rushabhT3

To edit commit messages, there is the `--amend` option of Git's `commit` command (`git commit`).  In a terminal of your choice, run the following instructions.

```bash
git commit --amend -m 'New Message for this Commit' # Enter your new commit message here.
git push --force # Push to the configured default remote repository for this branch.
```

With the `--amend` option, you will edit the latest commit, i.e. HEAD.  It is primarily intended to add changes to the last commit which one forgot to stage (`git add`) before committing.  This operation will revert the last commit, auto-stage its changes, and create a new commit by adding *all* staged changes to it.  When the new commit is created, you are allowed to alter the commit message which is exactly what you want to do.

By doing so, you will create a new commit which is not expected by your remote repository.  Your remote assumes that the old commit would still exist -- but it does not because you just replaced it with your edited commit.  Hence, you need to push the edited commit to *all* remotes by a force push (`git push --force`).  Make sure that the remote allows force pushes; see https://docs.github.com/en/repositories/configuring-branches-and-merges-in-your-repository/managing-protected-branches/about-protected-branches#allow-force-pushes for details.

Does this answer your question?  If I should not have understood your question correctly, please reply to this answer suggestion with a more detailed explanation of what your question is about.
